### ISSUE TITLE
Flac extension: Update build instructions to work with newer NDKs

### ISSUE THREAD

**Issue by motornyimaksym** [google/ExoPlayer#5601]
I follow this instruction and get https://github.com/google/ExoPlayer/tree/release-v2/extensions/flac.

```
Android NDK: android-14 is unsupported. Using minimum supported version android-16.    
Android NDK: WARNING: APP_PLATFORM android-16 is higher than android:minSdkVersion 1 in /home/maxim/StudioProjects/ExoPlayer/extensions/flac/src/main/AndroidManifest.xml. NDK binaries will *not* be compatible with devices older than android-16. See https://android.googlesource.com/platform/ndk/+/master/docs/user/common_problems.md for more information.    
/home/maxim/Android/Sdk/ndk-bundle/build/core/add-application.mk:178: *** Android NDK: APP_STL gnustl_static is no longer supported. Please switch to either c++_static or c++_shared. See https://developer.android.com/ndk/guides/cpp-support.html for more information.    .  Stop.
```

**Comment by ojw28** [comment]
What version of the Android NDK are you using? And what version of ExoPlayer are you using?

**Comment by motornyimaksym** [comment]
@ojw28 The latest one NDK version in stable Android Stuidio and Exoplayer from release-v2 branch.
The problem is here:
https://github.com/google/ExoPlayer/blob/release-v2/extensions/flac/src/main/jni/Application.mk
1. APP_PLATFORM should be at least android-16.
2. APP_STL should be changed to either c++_static or c++_shared.

**Comment by ojw28** [comment]
Thanks for the information!

@tonihei - Any chance you could take a look at this? I wonder if our other native extensions need any changes made to them too, since they seem to have similar `Application.mk` files? @andrewlewis may also know what needs to be done.

**Comment by tonihei** [comment]
The extension builds fine for me if I follow the steps in the README file. Note that the README explicitly asks for an NDK with version <= 17c. 

I suppose we should update our README to add build instructions for newer NDKs at some point. Let's use this issue to track this. Issue is also similar to #4940 and we should probably update all of them together.

**Comment by ojw28** [comment]
This was fixed in 2.11.0.
